docs/designs/link-output.md

```diff
@@ -0,0 +1,97 @@
+# Link And Bookmarks Output
+
+Docfx generate `links` and `bookmarks` information of each files into one single file `links.json` for future data utilization of downstream service of docfx.
+
+## Links
+
+`links` includes all the links which has been processed by docfx:
+
+- Relative link in source content will be processed by docfx and convert it to absolute link if it's valid, otherwise keep it untouched.
+
+- Absolute link in source content will NOT processed by docfx
+
+| Links in source content | Examples              | Output in `links.json` |
+| ----------------------- | --------------------- | ---------------------- |
+| Relative Link           | ./a.md (valid)        | /basepath/article/a    |
+|                         | a/b.md (invalid)      | a/b.md                 |
+| Absolute Link           | /basepath/articles/b  | /basepath/articles/b   |
+|                         | www.github.com/x      | www.github.com/x       |
+|                         | https://docs.com/y    | https://docs/com/y     |
+
+> NOTE: the absolute links converted from relative links by docfx don't contains `locale` and `versioning` information
+
+##  Bookmarks
+
+`bookmarks` includes all bookmarks ids generated/defined from source content, includes its inclusions(token/code-snippets)
+
+| Source content examples        | Html content                             | Output in `links.json` |
+| ------------------------------ | -----------------------------------------| ---------------------- |
+|  `# Title H1`                  | `<h1 id="title-h1">Title H1</h1>`        | title-h1               |
+|  `## Title H2`                 | `<h2 id="title-h2">Title H2</h2>`        | title-h2               |
+|  `## Title H2`                 | `<h2 id="title-h2">Title H2</h2>`        | title-h2               |
+| `<a id="##special-chars"></a>` | `<a id="##special-chars"></a>`           | ##special-chars        |
+| `<a id="with:colon"></a>`      | `<a id="with:colon"></a>`                | with:colon             |
+| `[!include[](d.md)]`           | `<h1 id="title-h1">Title H1</h1>`        | title-h1               |
+
+> The content in d.md is `# Title H1` in above case
+
+## Output
+
+`links.json` file will be generated by docfx to contains all the links and bookmarks information about all the page files in build.
+
+The output is **an array** of below data model
```

Comment:
consider making root level object instead of array, other it will be hard to insert a global scope data into it in the future.

Reply:
@yufeih